modules/acquia_cms_common/src/EventSubscriber/KernelTerminate/AcquiaCmsTelemetry.php

```diff
@@ -254,13 +282,83 @@ private function shouldSendTelemetryData(): bool {
     if (AcquiaDrupalEnvironmentDetector::isAhEnv()) {
       $telemetryOpted = $this->state->get('acquia_connector.telemetry.opted', TRUE);
       $isAcquiaTelemetryDataSent = $this->state->get('acquia_cms_telemetry.status', FALSE);
-      if ($telemetryOpted && !$isAcquiaTelemetryDataSent) {
+      if ($telemetryOpted && !$isAcquiaTelemetryDataSent && $this->shouldResendData()) {
         return TRUE;
       }
     }
+
     return FALSE;
   }
 
+  /**
+   * Decides if telemetry data should resend or not.
+   */
+  private function shouldResendData(): bool {
+    $telemetryDataOld = $this->state->get('acquia_cms_common.telemetry_data', []);
+    if (!empty($telemetryDataOld)) {
+      $telemetryDataOld = json_decode($telemetryDataOld, TRUE);
+    }
+    $telemetryDataCurrent = $this->getAcquiaCmsTelemetryData();
+    if (empty($telemetryDataOld) || (is_array($telemetryDataOld) &&
+    array_key_exists('acquia_cms', $telemetryDataOld) &&
+    array_diff_assoc($telemetryDataOld['acquia_cms'], $telemetryDataCurrent['acquia_cms']))) {
+      $this->state->set('acquia_cms_telemetry.status', FALSE);
+      return TRUE;
+    }
+    return FALSE;
+  }
+
+  /**
+   * Get an array of information about Lightning extensions.
+   *
+   * @return array
+   *   An array of extension info keyed by the extensions machine name. E.g.,
+   *   ['lightning_layout' => ['version' => '8.2.0', 'status' => 'enabled']].
+   */
+  private function getExtensionInfo() {
+    $all_modules = $this->moduleList->getAllAvailableInfo();
+    $installed_modules = $this->moduleList->getAllInstalledInfo();
+    $extension_info = [];
+    $acms_modules = [
+      'acquia_cms_article',
+      'acquia_cms_audio',
+      'acquia_cms_common',
+      'acquia_cms_component',
+      'acquia_cms_dam',
+      'acquia_cms_document',
+      'acquia_cms_event',
+      'acquia_cms_headless',
+      'acquia_cms_image',
+      'acquia_cms_page',
+      'acquia_cms_person',
+      'acquia_cms_place',
+      'acquia_cms_search',
+      'acquia_cms_site_studio',
+      'acquia_cms_starter',
+      'acquia_cms_tour',
+      'acquia_cms_toolbar',
+      'acquia_cms_video',
+      'sitestudio_config_management',
+    ];
+
+    foreach ($all_modules as $name => $extension) {
+      // Remove all custom modules from reporting.
+      if (strpos($this->moduleList->getPath($name), '/custom/') !== FALSE) {
+        continue;
+      }
+      if (in_array($name, $acms_modules)) {
+        // Version is unset for dev versions. In order to generate reports, we
+        // need some value for version, even if it is just the major version.
+        $extension_info[$name]['version'] = $extension['version'] ?? 'dev';
+
+        // Check if module is installed.
+        $extension_info[$name]['status'] = array_key_exists($name, $installed_modules) ? 'enabled' : 'disabled';
```

Comment:
$extension_info[$name]['status'] = array_key_exists($name, $this->moduleList->getAllInstalledInfo()) ? 'enabled' : 'disabled';

Reply:
Ohh my mistake, not required.